format_parser.gemspec

```diff
@@ -39,7 +39,11 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rspec', '~> 3.0'
   spec.add_development_dependency 'rake', '~> 12'
   spec.add_development_dependency 'simplecov', '~> 0.15'
-  spec.add_development_dependency 'pry', '~> 0.11'
+  if RUBY_VERSION > '2.4' && RUBY_ENGINE != 'jruby'
```

Comment:
This is executed at build time, which is always one of the maintainers' machines. I think you may want to conditionally load these gems at runtime, based on the runtime Ruby.

Reply:
> This is executed at build time, which is always one of the maintainers' machines.

yeah!

> I think you may want to conditionally load these gems at runtime, based on the runtime Ruby.

I'm not sure if I understood, but it doesn't matter if using `pry` or `pry-byebug`, you can use the same command as the breakpoint, which is `binding.pry`.

Do you have an example of a problem that could happen to do this way pls?

Reply:
Honestly, I think `pry` got into the gemspec by accident and it should be removed alltogether. Rationale:

* A dev dependency must be required to run tests or generate build products. Pry doesn't.
* A dev dependency must be used somewhere in the test/development infrastructure actively. Pry isn't.

The fact that pry merely "is going to be available" can even be considered a net negative, because when working on a gem it will stimulate the developer to add `binding.pry` somewhere and not notice that it actually requires pry to be set as a dependency. Given that I would prefer we _remove_ pry from the gemspec and leave it to the developer to require the variant of pry they prefer :-)

Reply:
ok, it makes sense! I've never thought in this way

By doing this, every time I want to use pry, I will need to use it like this?

```ruby
require 'pry' # or 'pry-byebug'
binding.pry
```

(I can edit my snippet to add the "require" line)

Reply:
Yes, but it will also fail CI build for you and thus remind that there is a stray `binding.pry` somewhere :-)

Reply:
humm, that's really a good way to protect us against that!

Reply:
So I created [this commit](https://github.com/WeTransfer/format_parser/pull/154/commits/b906535ae842520abce1727f3f07fb2ae879da99) removing `pry` and used some of your words to explain about it!

Reply:
Why not specify these kind of dependencies in the `Gemfile`? That's my approach when I don't want to pollute the gemspec with non-build/runtime-related dependencies.

Reply:
yeah, it's a good way!